common/src/main/java/com/microsoft/identity/common/internal/broker/AuthUxJavaScriptInterface.kt

```diff
@@ -0,0 +1,155 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.internal.broker
+
+import android.webkit.JavascriptInterface
+import com.google.gson.Gson
+import com.google.gson.JsonSyntaxException
+import com.google.gson.stream.MalformedJsonException
+import com.microsoft.identity.common.adal.internal.AuthenticationConstants
+import com.microsoft.identity.common.internal.numberMatch.NumberMatchHelper
+import com.microsoft.identity.common.logging.Logger
+
+/**
+ * JavaScript API to receive JSON string payloads from AuthUX in order to facilitate calling various
+ * broker methods.
+ */
+class AuthUxJavaScriptInterface {
+
+    // Store number matches in a static hash map
+    // No need to persist this storage beyond the current broker process, but we need to keep them
+    // long enough for AuthApp to call the broker api to fetch the number match
+    companion object {
+        val TAG = AuthUxJavaScriptInterface::class.java.simpleName
+        private const val JAVASCRIPT_INTERFACE_NAME = "ClientBrokerJS"
+
+        fun getInterfaceName() : String {
+            return JAVASCRIPT_INTERFACE_NAME
+        }
+
+        /**
+         * Helper method to determine if url is a valid Url for the JS Interface
+         * @param url url being loaded
+         * @return true if url is a valid, safe url, false otherwise
+         */
+        fun isValidUrlForInterface(url: String?): Boolean {
+            // If url is null, return false
+            if (url == null) {
+                return false
+            }
+
+            // Otherwise, make sure url is a valid url
+            // We only want to allow URLs that start with the AAD URL prefix or the MSA URL prefix
+            return url.startsWith(AuthenticationConstants.Broker.AAD_URL_PREFIX) ||
```

Comment:
Mentioned in another comment, probably better to convert String url to URL object and check the host part of the URL only for validation